qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,336 @@
+                on the method mentioned.
+            excitation_pool: An entire list of excitations.
+            threshold: the energy convergence threshold. It has a minimum value of 1e-15.
+            max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+        """
+        validate_min("threshold", threshold, 1e-15)
+
+        if adapt_gradient is None:
+            adapt_gradient = Gradient(grad_method="param_shift")
+        self._threshold = threshold
+        self.solver = solver
+        self._tmp_ansatz = self.solver.ansatz
+        self._max_iterations = max_iterations
+        self._adapt_gradient = adapt_gradient
+        self._excitation_pool = excitation_pool
+        self._excitation_list: List[OperatorBase] = []
+
+    @property
+    def initial_point(self) -> Optional[np.ndarray]:
+        """Returns initial point."""
+        return self.solver.initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: Optional[np.ndarray]) -> None:
+        """Sets initial point."""
+        self.solver.initial_point = initial_point
+
+    def _compute_gradients(
+        self,
+        theta: List[float],
+        operator: OperatorBase,
+    ) -> List[Tuple[float, PauliSumOp]]:
+        """
+        Computes the gradients for all available excitation operators.
+
+        Args:
+            theta: List of (up to now) optimal parameters
+            operator: operator whose gradient needs to be computed
+        Returns:
+            List of pairs consisting of gradient and excitation operator.
+        Raises:
+            AlgorithmError: If `quantum_instance` is not provided.
+        """
+        res = []
+        # compute gradients for all excitation in operator pool
+        if self.solver.quantum_instance is None:
+            raise AlgorithmError(
+                "A QuantumInstance or Backend must be supplied to run the quantum algorithm."
+            )
+        sampler = CircuitSampler(self.solver.quantum_instance)
+        for exc in self._excitation_pool:
+            # add next excitation to ansatz
+            self._tmp_ansatz.operators = self._excitation_list + [exc]
+            # the ansatz needs to be decomposed for the gradient to work
+            self.solver.ansatz = self._tmp_ansatz.decompose()
+            param_sets = list(self.solver.ansatz.parameters)
+            # zip will only iterate the length of the shorter list
+            parameter = dict(zip(self.solver.ansatz.parameters, theta))
+            op, expectation = self.solver.construct_expectation(
+                parameter, operator, return_expectation=True
+            )
+            # compute gradient
+            state_grad = self._adapt_gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            state_grad_result = sampler.convert(state_grad, params={param_sets[-1]: 0.0}).eval()
```

Comment:
Maybe we should rely on the value of `initial_point` for the given excitation index rather than defaulting to `0.0` here :thinking: 
I.e. do the following change:
```suggestion
            state_grad_result = sampler.convert(state_grad, params={param_sets[-1]: self.initial_point[idx]}).eval()
```
Along with the following change on line 123:
```diff
-         for exc in self._excitation_pool:
+         for idx, exc in enumerate(self._excitation_pool):
```

Reply:
What if the initial point is NoneType? Should I set the default initial point as 0.0?

Reply:
If `initial_point` is None, the `VQE` will overwrite it with a custom object of correct length. This is usually an array of random values in a given range (which can depend on the ansatz). So you should not worry about that here :+1: